File: keeping-up-with-the-javascripts/song.js
/*
  First Homework Assignment for the Pirple course Keeping Up with the Javascripts

  - Listing a song's characteristics using different Data types
*/

var songName = "Dance Monkey";
var artistName = "Tones and I";
var durationInSeconds = 237;
var genre = "Dance Pop";

var links = {
  // URL objects makes more sense then strings
  artist: new URL("https://www.tonesandi.com"),
  mailingList: new URL("https://bit.ly/2tL6gAW"),
  single: new URL("https://tonesandi.lnk.to/TheKidsAreComingYT"),
  album: new URL("https://tonesandi.lnk.to/DanceMonkeyYT"),
};

var management = {
  name: "Lemon Tree Music",
  email: "mgmt@tonesandi.com",
};

var followLinks = {
  facebook: new URL("https://www.facebook.com/TONESANDI"),
  instagram: new URL("https://www.instagram.com/tonesandi"),
};

var single = true;
var albumName = "The Kids Are Coming";

var musicVideo = {
  // Went for array when list of people
  directors: ["Liam Kelly", "Nick Kozakis"],
  directorPhotography: "Carl Allison",
  makeup: "Danielle Ruth",
  producer: "Visible Studios",
  producerLink: new URL("https://www.visiblestudios.com.au/"),
};

var youtube = {
  views: 853701861,
  subscribers: 2830000,
  likes: 8200000,
  dislikes: 374000,
  comments: 196598,
  channelLink: new URL ("http://smarturl.it/TonesAndI.YT"),
  // Date object made more sense then a string
  publishedDate: new Date('25 Jun 2019'),
  highestRatedComment: {
    likes: 12000,
    body: `
      Song : Move for me, Move for me, Move for me.


      Me : mufami mufami mufami.
    `,
    repliesCount: 288,
    // Date on the youtube video says 5 months ago
    postedDate: function () {
      var date = new Date('3 Apr 2020');
      date.setMonth(date.getMonth() - 5);
      return date;
    },
  },
};

// Found interpolating string syntax works with newlines and apostrophies
var lyrics = `
  They say oh my god I see the way you shine
  Take your hands my dear and place them both in mine
  You know you stopped me dead while I was passing by
  And now I beg to see you dance just one more time
  Ohhh I
  See you see you seen you every time
  and oh my I, I like you style
  You, you make me make me make me wanna cry
  And now I beg to see you dance just one more time

  So they say
  Dance for me
  Dance for me
  Dance for me
  Oh oh oh
  I’ve never seen anybody do the things you do before
  They say
  Move for me
  Move for me
  Move for me
  Eh eh eh
  And when you’re done i'll make you do it all again

  I said
  Oh my god I see you walking by
  Take my hands my dear and look me in my eyes
  Just like a monkey I’ve been dancing my whole life
  But you just beg to see me dance just one more time
  Ooh I
  See you see you seen you every time
  And oh my I, I like your style
  You, you make me make me make me wanna cry
  And now I beg to see you dance just one more time

  So they say
  Dance for me
  Dance for me
  Dance for me
  Oh oh oh
  I’ve never seen anybody do the things you do before
  They say
  Move for me
  Move for me
  Move for me
  Eh eh eh
  And when you’re done i'll make you do it all again
  They say
  Dance for me
  Dance for me
  Dance for me
  Oh oh oh, oh oh oh oh
  I’ve never seen anybody do the things you do before
  They say
  Move for me
  Move for me
  Move for me
  Eh eh eh
  And when you’re done i'll make you do it all again

  Oooooo

  They say
  Dance for me
  Dance for me
  Dance for me
  Oh oh oh
  I’ve never seen anybody do the things you do before
  They say
  Move for me
  Move for me
  Move for me
  Eh eh eh
  And when you’re done i'll make you do it all again
  They say
  Dance for me
  Dance for me
  Dance for me
  Oh oh oh, oh oh oh oh
  I’ve never seen anybody do the things you do before
  They say
  Move for me
  Move for me
  Move for me
  Eh eh eh
  And when you’re done i'll make you do it all again
  All again
`

console.log(songName)
console.log(artistName)
console.log(durationInSeconds)
console.log(genre)
console.log(links.artist)
console.log(links.mailingList)
console.log(links.single)
console.log(links.album)
console.log(management.name)
console.log(management.email)
console.log(followLinks.facebook)
console.log(followLinks.instagram)
console.log(single)
console.log(albumName)
console.log(musicVideo.directors)
console.log(musicVideo.directorPhotography)
console.log(musicVideo.makeup)
console.log(musicVideo.producer)
console.log(musicVideo.producerLink)
console.log(youtube.views)
console.log(youtube.subscribers)
console.log(youtube.likes)
console.log(youtube.dislikes)
console.log(youtube.comments)
console.log(youtube.channelLink)
console.log(youtube.publishedDate)
console.log(youtube.highestRatedComment.likes)
console.log(youtube.highestRatedComment.body)
console.log(youtube.highestRatedComment.repliesCount)
console.log(youtube.highestRatedComment.postedDate())
console.log(lyrics)

File: r-toolkit/song.r
#
#  First Homework Assignment for the Pirple course The R Toolkit
#
#  - Listing a song's characteristics using different Variable types
#

songName <- "Dance Monkey"
artistName <- "Tones and I"
durationInSeconds <- 237
genre <- "Dance Pop"

# Is the song published as a single?
single <- T

artistLink <- "https://www.tonesandi.com"
mailingListLink <- "https://bit.ly/2tL6gAW"
singleLink <- "https://tonesandi.lnk.to/TheKidsAreComingYT"
albumLink <- "https://tonesandi.lnk.to/DanceMonkeyYT"

managementName <- "Lemon Tree Music"
managementEmail <- "mgmt@tonesandi.com"

facebook <- "https://www.facebook.com/TONESANDI"
instagram <- "https://www.instagram.com/tonesandi"

albumName <- "The Kids Are Coming"

musicVideoDirectorPhotography <- "Carl Allison"
musicVideoMakeup <- "Danielle Ruth"
musicVideoProducer <- "Visible Studios"
musicVideoProducerLink <- "https://www.visiblestudios.com.au/"

youtubeViews <- 853701861
youtubeSubscribers <- 2830000
youtubeLikes <- 8200000
youtubeDislikes <- 374000
youtubeComments <- 196598
youtubeHighestRatedCommentLikes <- 12000

youtubeChannelLink <- "http://smarturl.it/TonesAndI.YT"
youtubePublishedDate <- '25 Jun 2019'

# Vector of the different youtube metrics
youtubeNumbersValues <- c(
  youtubeViews,
  youtubeSubscribers,
  youtubeLikes,
  youtubeDislikes,
  youtubeComments
)

youtubeNumbersHeadings <- c(
  "youtubeViews",
  "youtubeSubscribers",
  "youtubeLikes",
  "youtubeDislikes",
  "youtubeComments"
)

# Data frame of Headings and Values of youtube metrics
youtubeNumbers <- data.frame(youtubeNumbersHeadings, youtubeNumbersValues)

# Some vector operations on the different youtube metrics
youtubeNumbersInThousands <- youtubeNumbersValues / 1000
youtubeNumbersInMillions <- youtubeNumbersInThousands / 1000
areGreaterThenOneMillion <- youtubeNumbersInMillions >= 1

# Which of the youtube metrics are greater then one million as a vector
youtubeNumbersInMillionsGreaterThenOneMillionVector <- youtubeNumbersInMillions[areGreaterThenOneMillion]

# Which of the youtube metrics are greater then one million as a data frame of the headings
youtubeNumbersInMillionsGreaterThenOneMillionHeadings <- youtubeNumbers[youtubeNumbers$youtubeNumbersValues >= 1000000,][c(1)]

youtubeHighestRatedCommentBody <- "\n\
  Song : Move for me, Move for me, Move for me.\n\
  \n\
  Me : mufami mufami mufami.\n\
"
youtubeHighestRatedCommentRepliesCount <- 288

# List item of the lines in the lyrics
lyrics <- list(
  "\n",
  "They say oh my god I see the way you shine\n",
  "Take your hands my dear and place them both in mine\n",
  "You know you stopped me dead while I was passing by\n",
  "And now I beg to see you dance just one more time\n",
  "Ohhh I\n",
  "See you see you seen you every time\n",
  "and oh my I, I like you style\n",
  "You, you make me make me make me wanna cry\n",
  "And now I beg to see you dance just one more time\n",
  "\n",
  "So they say\n",
  "Dance for me\n",
  "Dance for me\n",
  "Dance for me\n",
  "Oh oh oh\n",
  "I’ve never seen anybody do the things you do before\n",
  "They say\n",
  "Move for me\n",
  "Move for me\n",
  "Move for me\n",
  "Eh eh eh\n",
  "And when you’re done i'll make you do it all again\n",
  "\n",
  "I said\n",
  "Oh my god I see you walking by\n",
  "Take my hands my dear and look me in my eyes\n",
  "Just like a monkey I’ve been dancing my whole life\n",
  "But you just beg to see me dance just one more time\n",
  "Ooh I\n",
  "See you see you seen you every time\n",
  "And oh my I, I like your style\n",
  "You, you make me make me make me wanna cry\n",
  "And now I beg to see you dance just one more time\n",
  "\n",
  "So they say\n",
  "Dance for me\n",
  "Dance for me\n",
  "Dance for me\n",
  "Oh oh oh\n",
  "I’ve never seen anybody do the things you do before\n",
  "They say\n",
  "Move for me\n",
  "Move for me\n",
  "Move for me\n",
  "Eh eh eh\n",
  "And when you’re done i'll make you do it all again\n",
  "They say\n",
  "Dance for me\n",
  "Dance for me\n",
  "Dance for me\n",
  "Oh oh oh, oh oh oh oh\n",
  "I’ve never seen anybody do the things you do before\n",
  "They say\n",
  "Move for me\n",
  "Move for me\n",
  "Move for me\n",
  "Eh eh eh\n",
  "And when you’re done i'll make you do it all again\n",
  "\n",
  "Oooooo\n",
  "\n",
  "They say\n",
  "Dance for me\n",
  "Dance for me\n",
  "Dance for me\n",
  "Oh oh oh\n",
  "I’ve never seen anybody do the things you do before\n",
  "They say\n",
  "Move for me\n",
  "Move for me\n",
  "Move for me\n",
  "Eh eh eh\n",
  "And when you’re done i'll make you do it all again\n",
  "They say\n",
  "Dance for me\n",
  "Dance for me\n",
  "Dance for me\n",
  "Oh oh oh, oh oh oh oh\n",
  "I’ve never seen anybody do the things you do before\n",
  "They say\n",
  "Move for me\n",
  "Move for me\n",
  "Move for me\n",
  "Eh eh eh\n",
  "And when you’re done i'll make you do it all again\n",
  "All again\n"
)
